components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -5772,7 +5772,10 @@ public ApiTypeWrapper getAPIorAPIProductByUUID(String uuid, String organizationI
                 String msg = "Failed to get API. API artifact corresponding to artifactId " + uuid + " does not exist";
                 throw new APIMgtResourceNotFoundException(msg);
             }
-        } catch (APIPersistenceException | OASPersistenceException | ParseException e) {
+        } catch (APIPersistenceException e) {
+            String msg = "Failed to get API. API artifact corresponding to artifactId " + uuid + " does not exist";
+            throw new APIMgtResourceNotFoundException(msg);
```

Comment:
Can't we throw an APIManagementException with the exception in [1] so that we can directly pass this error to the Global thrower?

[1] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/ExceptionCodes.java#L120